build-support/bin/deploy_to_s3.py

```diff
@@ -12,21 +12,21 @@
 def main() -> None:
     if shutil.which("aws") is None:
         install_aws_cli()
-    setup_authentication()
+    validate_authentication()
     deploy()
 
 
 def install_aws_cli() -> None:
     subprocess.run(["./build-support/bin/install_aws_cli_for_ci.sh"], check=True)
 
 
-def setup_authentication() -> None:
-    access_key_id = "AWS_ACCESS_KEY_ID__TO_BE_REEXPORTED_ON_DEPLOYS"
```

Comment:
No more need for the `AWS_ACCESS_KEY_ID__TO_BE_REEXPORTED_ON_DEPLOYS` nonsense. We use GitHub secrets to only set the env var when we know we're doing a deploy.